include/nbl/video/utilities/IGPUObjectFromAssetConverter.h

```diff
@@ -1685,72 +1694,90 @@ inline created_gpu_object_array<asset::ICPUDescriptorSet> IGPUObjectFromAssetCon
+                                info->info.buffer.size = descriptorInfos[offset+d].buffer.size;
                             }
                             else
                             {
-                                const auto imageFormat = static_cast<asset::ICPUImageView*>(info->desc.get())->getCreationParameters().format;
-                                info->image.imageLayout = isDepthOrStencilFormat(imageFormat) ? asset::IImage::EL_DEPTH_STENCIL_READ_ONLY_OPTIMAL : asset::IImage::EL_SHADER_READ_ONLY_OPTIMAL;
-
-                                if (desc.image.sampler)
-							        info->image.sampler = gpuSamplers->operator[](smplrRedirs[si++]);
+                                info->desc = nullptr;
+                                info->info.buffer.offset = 0u;
+                                info->info.buffer.size = 0u;
                             }
+                            ++bi;
                         }
-					}
-                    allDescriptorsPresent = allDescriptorsPresent && info->desc;
-					info++;
-				}
-                if (allDescriptorsPresent)
-                    write_it++;
-			}
+                        else if (isBufviewDesc(type))
+                        {
+                            info->desc = bufviewRedirs[bvi] >= gpuBufviews->size() ? nullptr : gpuBufviews->operator[](bufviewRedirs[bvi]);
+                            ++bvi;
+                        }
+                        else if (isSampledImgViewDesc(type) || isStorageImgDesc(type))
+                        {
+                            info->desc = imgViewRedirs[ivi] >= gpuImgViews->size() ? nullptr : gpuImgViews->operator[](imgViewRedirs[ivi]);
+                            ++ivi;
+                            // TODO: This should be set in the loader (or whoever is creating
+                            // the descriptor)
+                            if (info->info.image.imageLayout == asset::IImage::EL_UNDEFINED)
+                            {
+                                if (isStorageImgDesc(type))
+                                {
+                                    info->info.image.imageLayout = asset::IImage::EL_GENERAL;
+                                }
+                                else
+                                {
+                                    const auto imageFormat = static_cast<asset::ICPUImageView*>(info->desc.get())->getCreationParameters().format;
+                                    info->info.image.imageLayout = isDepthOrStencilFormat(imageFormat) ? asset::IImage::EL_DEPTH_STENCIL_READ_ONLY_OPTIMAL : asset::IImage::EL_SHADER_READ_ONLY_OPTIMAL;
+
+                                    if (samplers.size() != 0ull)
+                                        info->info.image.sampler = gpuSamplers->operator[](smplrRedirs[si++]);
+                                }
+                            }
```

Comment:
take the `imageLayout` from the `ICPUDescriptorSetLayout` , stop doing this check.

or whenever the layout is not undefined nothing will get assigned to the sampler.


Or fix it up some other way.

Reply:
I guess we're not done with this one yet

Reply:
Resolved.